### PR TITLE
Add 'fluent' enum to teacher's English proficiency 

### LINKED ISSUE #85: Update Teacher's language proficiency enum
Unlike for students, the language proficiency variants for teachers should be as below. This should be updated in the teachers migration as well as `NewTeacher.vue` and relevant frontend components.

```
Little (Able to understand simple words)
Simple (Able to speak full sentences)
Intermediate (Able to understand simple words)
Fluent (Able to engage in a conversation)
```

(from Figma)
image

### PR DESCRIPTION
**Pull request by dzldebby**
Closes #85

### CODE REVIEW

**Comment by dzldebby** [comment]
Original: image

Suggested: image

We made the assumption that 'Language Proficiency' on the 'NewTeacher.vue' refers to Native Language and not the Second Language. 

Suggested changes:
1. Re-named 'Language Proficiency' to 'Native Language Proficiency' 
2. Re-aligned dropdown fields of 'NewTeacher.vue'
3. Shifted 'English Proficiency' field to its own row between 'Source' and 'Native Language'

File: backend/migrations/20210214110931_teachersProficiency.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  };
+
+  exports.down = async function(knex) {
+    return knex.raw(`
+        ALTER TABLE "teachers" DROP CONSTRAINT "teachers_LanguageProficiency_check";
+        ALTER TABLE "teachers" ADD CONSTRAINT "teachers_LanguageProficiency_check"
+        CHECK ("LanguageProficiency" IN ('Little', 'Simple', 'Intermediate', 'Fluent'));
+        ALTER TABLE "teachers" DROP CONSTRAINT "teachers_EnglishProficiency_check";
+        ALTER TABLE "teachers" ADD CONSTRAINT "teachers_EnglishProficiency_check"
+        CHECK ("EnglishProficiency" IN ('Little', 'Simple', 'Intermediate', 'Fluent'));
```
Comment (thchia) [inline backend/migrations/20210214110931_teachersProficiency.js]: This should be the opposite of what is defined in the "up" function. So I think here you should be removing the constraint?

File: frontend/src/pages/NewTeacher.vue [hunk 2]
```diff
@@ -226,7 +229,7 @@ export default {
         NativeLanguageString: this.selected.NativeLanguage,
         SecondLanguageString: this.selected.SecondLanguage,
         EnglishProficiency: this.EnglishProficiency,
-        LanguageProficiency: this.SecondLanguageProficiency,
+        LanguageProficiency: this.NativeLanguageProficiency,
```
Comment (thchia) [inline frontend/src/pages/NewTeacher.vue]: Is there a reason you changed it from `SecondLanguageProficiency` to `NativeLanguageProficiency`? The way I understand it, a Teacher has the following properties:

- Native Language (with no proficiency, as you should be proficient in your native language)
- Second Language
- Second Language Proficiency
- English Proficiency (where even if both the Teacher's Native and Second languages are **not** English, we still store this)

**Review by thchia — CHANGES_REQUESTED** [review]
Left some comments

File: frontend/src/pages/NewTeacher.vue (hunk 2 above)
Comment (therealyingtong) [reply frontend/src/pages/NewTeacher.vue]: oh that makes sense. we were interpreting Native Language to have a proficiency. we'll revert this change.

File: backend/migrations/20210214110931_teachersProficiency.js (hunk 1 above)
Comment (therealyingtong) [inline backend/migrations/20210214110931_teachersProficiency.js]:
```suggestion
        ALTER TABLE "teachers" DROP CONSTRAINT "teachers_LanguageProficiency_check";
        ALTER TABLE "teachers" DROP CONSTRAINT "teachers_EnglishProficiency_check";
```

Comment (therealyingtong) [reply backend/migrations/20210214110931_teachersProficiency.js]: the `ADD CONSTRAINT` operations have now been removed.

**Review by therealyingtong — APPROVED** [review]

**Review by thchia — APPROVED** [review]

**Review by lauratsui — APPROVED** [review]
LGTM

**Pull request merged**